fixed #2

bsutton opened the pull request.

bsutton committed:
added logic to deal with hosted dependencies where the version is in the yaml before the hosted key. The pub get command supports this odd order and so must wek.

bsutton committed:
Fixed #2. The executable script now returns the correct path even if the script value in pubspec.yaml is empty.

bsutton commented:
@j4qfrost just a ping.

Are you able to merge this PR.
If it helps, I believe I still have the permissions required to publish and can do so post-merge.

bsutton committed:
ignored .failed_tracker

bsutton committed:
Released 2.2.0.

bsutton committed:
Released 2.2.0.

bsutton commented:
@j4qfrost I've gone ahead and published this to pub.dev as I need the fix urgently.

If you don't have time for maintenance on this project I would be happy to take over as it's now a major piece of a project I'm working on.

Otherwise can you merge when you have a chance so the main branch reflects the current 2.2.0 release.

j4qfrost commented on test/executable_test.dart:
```diff
@@ -1,8 +1,9 @@
 
 main() {
-  test('executables ', () {
+  test('executables ', () async {
```
Not necessary.

j4qfrost commented on lib/src/executable.dart:
```diff
@@ -12,12 +12,11 @@ import 'json_utils.dart';
 
 class Executable extends Jsonable {
-  // String name;
+  String name;
```
Just change `script` to a non-nullable and handle everything that way if you think it's necessary. `name` is just extra stuff.

bsutton replied:
name and script can be different and having both in the class seems to be handy rather than requiring the user to have access to the key to get the name.

j4qfrost replied:
This changes the behavior of the constructor across multiple different projects for one. `script` already suffices and yaml allows for nulls, so the behavior is consistent there. Users should just account for scenarios where there is no string provided or we should just force the behavior regardless. I don't actually know your specific use case, but this seems bad.

bsutton committed:
removed an unused async in a unit test.

bsutton commented:
To be honest, I hadn't actually considered the ctor to be part of the
public interface, but you are correct it is.

The problem is that the current API is broken. The name and script can be
set independently which the original version of the class didn't reflect.

In reality, I doubt that this class is being used as you would have to be
building a pubspec from scratch to consider using it and the code that I've
seen that builds a pubspec from scratch use templates rather than the
pubspec api.

My vote would be to make the change. If necessary we increment the major
version no. to reflect a break in the api.

S. Brett Sutton
Noojee Contact Solutions
03 8320 8100


On Sun, 5 Jun 2022 at 15:18, j4qfrost ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In lib/src/executable.dart
> <https://github.com/j4qfrost/pubspec/pull/6#discussion_r889650510>:
>
> > @@ -12,12 +12,11 @@ import 'json_utils.dart';
>  /// typing <name> executes bin/<script>.dart.
>
>  class Executable extends Jsonable {
> -  // String name;
> +  String name;
>
> This changes the behavior of the constructor across multiple different
> projects for one. script already suffices and yaml allows for nulls, so
> the behavior is consistent there. Users should just account for scenarios
> where there is no string provided or we should just force the behavior
> regardless. I don't actually know your specific use case, but this seems
> bad.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/j4qfrost/pubspec/pull/6#discussion_r889650510>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAG32OCLTXDVFPZ4B65WVZTVNQ2CFANCNFSM5XDQP7CQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

bsutton committed:
Added support for the platforms tag. The current system would read the platforms keyword but when writting would ouput the likes of 'linux: null'. pub.dev does not like the null.

bsutton committed:
Released 2.3.0.

bsutton committed:
added transient dependency dcli_core to satisfy pub publish.

bsutton committed:
Fixed a critical bug in the save method. We were failing to flush the writes before closing the ioSink with the result that sometimes only a partial pubspec.yaml was written leaving the project in a corrupt state.

bsutton committed:
Release notes for 2.4.0

bsutton committed:
Added a repository keyword to the pubspec.yaml so its more obvious to users looking for the git repo on pub.dev.

bsutton commented:
@j4qfrost can we reach a conclusion on this.
I've just raised merged another commit into the PR as I've found a critical bug in save method which results in a truncated pubspec.yaml.

bsutton committed:
Released 2.4.0.

bsutton committed:
Released 2.4.0.

bsutton committed:
Fixed a bug in the initialisation of an executable script name. Added unit tests for same.

bsutton committed:
changed the barrel file name to pubspec2 to confirm with dart conventions.

bsutton committed:
Released 2.4.1.

bsutton committed:
Released 2.4.2.

bsutton committed:
improvements to the type system

bsutton committed:
enforced typing across the package.
Fixed a bug on windows where the executable script path used the wrong delimiter.

bsutton committed:
Released 2.5.0.

bsutton committed:
moved dependency types into their own libraries.

bsutton committed:
Released 3.0.0.

bsutton committed:
Released 3.0.0.

bsutton committed:
Released 4.0.0.
